crates/flux-driver/src/collector/mod.rs

```diff
@@ -536,6 +542,102 @@ impl<'a, 'tcx> SpecCollector<'a, 'tcx> {
             self.specs.infer_opts.insert(def_id, infer_opts);
         }
     }
+
+    fn collect_detached_specs(&mut self, attrs: &mut FluxAttrs, def_id: LocalDefId) -> Result {
```

Comment:
I'd put the stuff related to detached specs in a separate module (like we do for extern specs). This will probably become complicated soon.